src/compiler/parser.ts

```diff
@@ -9226,6 +9215,24 @@ namespace Parser {
                 return finishNode(create(name, text.join("")), start, scanner.getTokenEnd());
             }
 
+            function parseJSDocLinkName() {
+                if (tokenIsIdentifierOrKeyword(token())) {
+                    const pos = getNodePos();
+
+                    let name: EntityName | JSDocMemberName = parseIdentifierName();
+                    while (parseOptional(SyntaxKind.DotToken)) {
+                        name = finishNode(factory.createQualifiedName(name, token() === SyntaxKind.PrivateIdentifier ? createMissingNode<Identifier>(SyntaxKind.Identifier, /*reportAtCurrentPosition*/ false) : parseIdentifier()), pos);

```

Comment:
the key difference in the new code is that it's a choice between parseIdentifier and createMissingNode. In the old code, parseRightSideOfDot chooses between five different functions. That seems better, but obviously it was wrong. How was it wrong?

Reply:
@sandersn Thanks for the review. `parseRightSideOfDot` always throws an error for private identifiers that should be allowed in this case. (`/*reportAtCurrentPosition*/ true, Diagnostics.Identifier_expected`). There is an option to control this with a new parameter, which requires a deep function chain change, or just use `createMissingNode` directly

https://github.com/microsoft/TypeScript/blob/5f5ed0f487d43127257154e999eefc1f25208368/src/compiler/parser.ts#L3611-L3613

https://github.com/microsoft/TypeScript/blob/5f5ed0f487d43127257154e999eefc1f25208368/src/compiler/parser.ts#L2598-L2603